src/fairseq2/datasets/huggingface.py

```diff
@@ -4,29 +4,25 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
-from typing import Any, Callable, Optional, Union
+from __future__ import annotations
+
+from typing import Any, Callable, Dict, Optional, Union
+
+from datasets import (  # type: ignore[attr-defined,import-untyped,import-not-found]
+    Dataset,
+    DatasetDict,
+)
 
 from fairseq2.data.data_pipeline import Collater, create_bucket_sizes, read_sequence
 from fairseq2.datasets.batching import LengthBatching, StaticBatching
 from fairseq2.datasets.data_reader import BatchT, DataPipelineReader
 from fairseq2.gang import Gang
 
-try:
```

Comment:
hmm, why do we want to remove this safeguard here ?

Although we have implemented the similar safeguard in fairseq2 CLI, this package in `fairseq2.datasets` can be consumed by other non-CLI packages / client too.

Reply:
Alright!. I have reverted this back https://github.com/facebookresearch/fairseq2/pull/708/commits/6bb31febe0b67f061bc8dca8179d0580751abc6f